packages/client/src/__tests__/integration/happy/rejectNotFound/test.ts

```diff
@@ -50,21 +50,42 @@ for (const constructorKey of Object.keys(cases.constructor)) {
       const value = currentMethod[valueKey]
       test(`rejectOnNotFound | constructor=${constructorKey} | ${method}=${value}`, async () => {
         // It should fail or not
-        expect.assertions(1)
         const PrismaClient = await getTestClient()
         const prisma = new PrismaClient({
           rejectOnNotFound: constructor,
         })
 
-        // Test Rejection
-        try {
+        // This function name is important cause we're searching
+        // for it's name in the stack bellow
+        const testRejectionOnNotFound = async () => {
           const r = await prisma.user[method]({
             where: { id: 'none' },
             rejectOnNotFound: value,
           })
-          expect(r).toMatchSnapshot()
+
+          // If it got to here, then no error was thrown
+          // so we check the value if it's equal to null
+          expect(r).toBeNull()
+        }
+
+        try {
+          await testRejectionOnNotFound()
         } catch (error) {
-          expect(error).toMatchSnapshot()
+          const { message, stack }: { message: string; stack: string } = error
+          expect(stack).toBeDefined()
+          expect(message).toBeDefined()
+
+          // Checking for the most important parts in the stack trace
+          // 1. The "ErrorName: ErrorMsg"
+          // 2. The function name that the error was thrown from
+          const isCustomError = error.name === 'NotFoundError'
+
+          if (isCustomError) {
+            expect(stack.includes(`NotFoundError: ${error.message}`)).toBeTruthy()
+          } else {
+            expect(stack.includes(`Error: ${error.message}`)).toBeTruthy()
+          }
```

Comment:
I don't think checking this provides a lot of value since it's testing V8 instead of Prisma, basically. It will also make things more complicated if/when there will be more error types. Just checking `expect(stack).toContain('at testRejectionOnNotFound')` should be enough.

Instead of this, I would prefer to have a snapshot test for the error message, and get the removed snapshots back:

```suggestion
          expect(error).toMatchSnapshot()
```

Reply:
I think jest serializes errors with stack traces and these differ on OSes. Can you confirm this @YassinEldeeb ?

Reply:
@aqrln yeah you're right, that's just extra unnecessary checks that can break in the future

Reply:
> Instead of this, I would prefer to have a snapshot test for the error message, and get the removed snapshots back:

but the old snapshot have some unnecessary values that equal null that has been replaced with [this line](https://github.com/prisma/prisma/pull/10645/files#diff-6b9e6e1a274c24674bd0fda583cc7f9972b206db55cc958f6f9b5e03f510dddaR68) , so I think It's better updating the snapshot to only include this `expect(error).toMatchSnapshot()` so It looks like this

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rejectOnNotFound | constructor=customError | findFirst=() => new Error('FindFirst Custom Error') 1`] = `FindFirst Custom Error`;

exports[`rejectOnNotFound | constructor=customError | findFirst=true 1`] = `No User found`;

exports[`rejectOnNotFound | constructor=customError | findFirst=undefined 1`] = `Constructor Custom Error`;

exports[`rejectOnNotFound | constructor=customError | findUnique=() => new Error('FindUnique Custom Error') 1`] = `FindUnique Custom Error`;

exports[`rejectOnNotFound | constructor=customError | findUnique=true 1`] = `No User found`;

...
```

Reply:
@YassinEldeeb yes, precisely. Sorry, I should have made myself more clear.

Reply:
Also, to clarify just in case, `expect(stack).toContain('at testRejectionOnNotFound')` check should still be there separately.

Reply:
I've committed the checking for a snapshot instead of all of these unnecessary shenanigans

[commit](https://github.com/prisma/prisma/pull/10645/commits/7cd800ad34a7486a0350fffc3fb3b695b64d7bfe)